app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Users
+  class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+    def google_oauth2
+      if user.persisted?
```

Comment:
Can the user be not persist in this controller? If yes, in which scenario?

Reply:
I can imagine such the outcome, but we've already discussed the subject in depth. It's removed.